src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -21,120 +103,808 @@ const ActorInit En_Pamera_InitVars = {
+    this->actionFunc = func_80BD8D1C;
+}
+
+void func_80BD8D1C(EnPamera* this, GlobalContext* globalCtx) {
+    s32 temp_v0 = (this->unk_31C < 0xC9) ^ 1;
+
+    this->unk_31C++;
+    if (temp_v0) {
+        this->actor.speedXZ = 1.5f;
+        func_800BDC5C(&this->skelAnime, animations, 1);
+        func_80BD8D80(this);
+    }
+}
+
+void func_80BD8D80(EnPamera* this) {
+    if (this->pathIndex > 0) {
+        this->pathIndex--;
+    }
+    this->actor.gravity = -2.0f;
+    this->actionFunc = func_80BD8DB0;
+}
+
+void func_80BD8DB0(EnPamera* this, GlobalContext* globalCtx) {
+    Vec3f vec;
+    s16 sp32;
+
+    vec.x = this->pathPoints[this->pathIndex].x;
+    vec.y = this->pathPoints[this->pathIndex].y;
+    vec.z = this->pathPoints[this->pathIndex].z;
+    sp32 = Math_Vec3f_Yaw(&this->actor.world.pos, &vec);
+    if (Math_Vec3f_StepToXZ(&this->actor.world.pos, &vec, this->actor.speedXZ) > 10.0f) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, sp32, 0xA, 0x3000, 0x100);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    } else {
+
+        if (this->pathIndex > 0) {
+            this->pathIndex--;
+        } else {
+            func_80BD9338(this, globalCtx);
+            func_80BD8A38(this);
+        }
+    }
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4U);
+    this->actor.world.pos.y += this->actor.gravity;
+}
+
+void func_80BD8F0C(EnPamera* this) {
+    func_80BD93CC(this, 1, 1);
+    func_800BDC5C(&this->skelAnime, animations, 4);
+    this->actionFunc = func_80BD8F60;
+}
+
+void func_80BD8F60(EnPamera* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xA, 0x3000, 0x1000);
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
+        func_800BDC5C(&this->skelAnime, animations, 2);
+        this->actor.speedXZ = 3.0f;
+        func_80BD93CC(this, 0, 0);
+        func_80BD8D80(this);
+    }
+}
+
+void func_80BD8FF0(EnPamera* this) {
+    Vec3f pameraPos;
+    s16 pameraYaw;
+
+    this->pathIndex = this->pathPointsCount - 1;
+    Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->pathPoints[this->pathPointsCount - 1]);
+    Math_Vec3s_ToVec3f(&pameraPos, &this->pathPoints[this->pathPointsCount - 2]);
+    pameraYaw = Math_Vec3f_Yaw(&pameraPos, &this->actor.world.pos);
+    this->actor.shape.rot.y = pameraYaw;
+    this->actor.world.rot.y = pameraYaw;
+    func_800BDC5C(&this->skelAnime, animations, 3);
+    this->actionFunc = func_80BD909C;
+}
+
+void func_80BD909C(EnPamera* this, GlobalContext* globalCtx) {
+}
+
+void func_80BD90AC(EnPamera* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK && (this->actionFunc != func_80BD8758) &&
+        (this->actionFunc != func_80BD8964) && (this->actionFunc != func_80BD8A7C) &&
+        (this->actionFunc != func_80BD8F60) && ((this->actionFunc != func_80BD8B70) || (this->pathIndex)) &&
```

Comment:
```suggestion
        (this->actionFunc != func_80BD8F60) && ((this->actionFunc != func_80BD8B70) || (this->pathIndex != 0)) &&
```
Since `this->pathIndex` isn't a boolean, I'd prefer the explicit check to non zero.